bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -1972,4 +2007,33 @@ public void closeComplete(int rc, LedgerHandle lh, Object ctx) {
         // becomes a property of the LedgerHandle itself.
         return LedgerMetadataUtils.getCurrentEnsemble(versionedMetadata.getValue());
     }
+
+    /**
+     * Return a {@link WriteSet} suitable for reading a particular entry.
+     * This will include all bookies that are cotna
+     */
+    WriteSet getWriteSetForReadOperation(long entryId) {
+        if (stickyBookieIndex != -1) {
+            // When sticky reads are enabled we want to make sure to take
+            // advantage of read-ahead (or, anyway, from efficiencies in
+            // reading sequential data from disk through the page cache).
+            // For this, all the entries that a given bookie prefetches,
+            // should read from that bookie.
+            // For example, with e=2, w=2, a=2 we would have
+            // B-1 B-2
+            // e-0 X X
+            // e-1 X X
+            // e-2 X X
+            //
+            // In this case we want all the requests to be issued to B-1 (by
+            // preference), so that cache hits will be maximized.
+            //
+            // We can only enable sticky reads if the ensemble==writeQuorum
+            // otherwise the same bookie will not have all the entries
+            // stored
+            return distributionSchedule.getWriteSet(stickyBookieIndex);
```

Comment:
Is this an entryId or a reference to a bookie.
I am a bit confused

Reply:
The write set is typically dependent on the entry id. If a `stickyBookieIndex` was set, it means we're try to override that distribution and instead send all reads to a particular bookie. `Bookie-i ` will be the "master" bookie for the `entry-i`. Therefore, a write set with `bookie-i` will first try to read from `bookie-i`, falling back to other bookies in case of failure.

Reply:
@eolivelli Does this answer the question?

Reply:
Okay. It works.
Sorry for late reply